Open Judge System/OJS.Services/Business/ProblemGroups/ProblemGroupsBusinessService.cs

```diff
@@ -40,27 +40,18 @@ public ServiceResult DeleteById(int id)
             return ServiceResult.Success;
         }
 
-        public void DeleteByProblem(int problemId)
-        {
-            var problemGroup = this.problemGroupsData.GetByProblem(problemId);
-
-            if (problemGroup == null || problemGroup.Problems.Any(p => !p.IsDeleted))
-            {
-                return;
-            }
-
-            this.Delete(problemGroup);
-        }
-
         public void DeleteByContest(int contestId)
```

Comment:
Shouldn't the CRUD-related functionality (Delete in this case) reside in the Data rather than Business service?

Reply:
In this case the process of deletion requires some business logic also, except just deleting the entities.